db/repositories/run_repository.py

```
@@ -0,0 +1,113 @@
+class SQLiteRunRepository(RunRepository):
+    """SQLite implementation of RunRepository.
+    
+    Uses functions from db.db module to interact with SQLite database.
+    """
+    
+    # Valid state transitions for run status
+    VALID_TRANSITIONS = {
+        RunStatus.RUNNING: {RunStatus.COMPLETED, RunStatus.FAILED},
+        RunStatus.COMPLETED: set(),  # Terminal state
+        RunStatus.FAILED: set(),     # Terminal state
+    }
```

Ruff (0.14.6)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)